src/components/admin/Settings/StorageMappings/Dialog/index.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+    Box,
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    IconButton,
+    Stack,
+    Typography,
+    useTheme,
+} from '@mui/material';
+import StorageMappingForm from 'components/admin/Settings/StorageMappings/Dialog/Form';
+import RepublicationLogs from 'components/admin/Settings/StorageMappings/Dialog/Logs';
+import ProviderSelector from 'components/admin/Settings/StorageMappings/Dialog/ProviderSelector';
+import SaveButton from 'components/admin/Settings/StorageMappings/Dialog/SaveButton';
+import { useStorageMappingStore } from 'components/admin/Settings/StorageMappings/Store/create';
+import Error from 'components/shared/Error';
+import ExternalLink from 'components/shared/ExternalLink';
+import { Cancel } from 'iconoir-react';
+import { Dispatch, SetStateAction } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+interface Props {
+    headerId: string;
+    open: boolean;
+    selectedTenant: string;
+    setOpen: Dispatch<SetStateAction<boolean>>;
+}
+
+const TITLE_ID = 'configure-storage-dialog-title';
+
+const docsUrl =
+    'https://docs.estuary.dev/getting-started/installation/#configuring-your-cloud-storage-bucket-for-use-with-flow';
+
+function ConfigureStorageDialog({
+    headerId,
+    open,
+    selectedTenant,
+    setOpen,
+}: Props) {
+    const theme = useTheme();
+
+    const logToken = useStorageMappingStore((state) => state.logToken);
+    const resetState = useStorageMappingStore((state) => state.resetState);
+    const serverError = useStorageMappingStore((state) => state.serverError);
+    const saving = useStorageMappingStore((state) => state.saving);
+    const setSaving = useStorageMappingStore((state) => state.setSaving);
+
+    const closeDialog = (event: React.MouseEvent<HTMLElement>) => {
+        event.preventDefault();
+
+        setOpen(false);
+        resetState();
+        setSaving(false);
+    };
+
+    return (
```

Comment:
Very small improvements - but since this is all store based can you break this dialog up a bit so that chunks that do not need to re-render are not re-rendered.

Probably just title, content , and actions